### DISCUSSION TITLE
Is there a way to insert the latest 3 blog posts on my home page?

### DISCUSSION THREAD

**Discussion by MrMacStripe** [datenstrom/community#79] [Ask a question] [answered]
Hi, is there a way to display the latest three blog posts on my One-Page-Layout home site? The /blog/ site has to be available, too. I think that should be easy to do, but I don´t see how to do it.

Thank you in advance!

**Comment by markseuffert** [comment]
Hi, you can add `[blogrecent /blog/ 3]` to your home page. Is that what you're looking for?

**Comment by MrMacStripe** [comment]
This adds a list to my page. I need something like a preview as you can create with wordpress for example: headline, image, first paragraph of the latest three articles. Basically it should show exactly what is shown at /blog/, but just the latest

**Comment by markseuffert** [comment, marked as answer]
Yes that's possible. Here's an example layout file `system/layouts/home.html`:

```
<?php $this->yellow->layout("header") ?>
<?php $pages = $this->yellow->content->index()->filter("layout", "blog")->sort("published", false)->limit(3) ?>
<?php $this->yellow->page->setLastModified($pages->getModified()) ?>
<?php $this->yellow->page->setHeader("Cache-Control", "max-age=60") ?>
<div class="content">
<div class="main" role="main">
<?php echo $this->yellow->page->getContent() ?>
<?php foreach($pages as $page): ?>
<div class="entry">
<div class="entry-title"><h1><a href="<?php echo $page->getLocation(true) ?>"><?php echo $page->getHtml("title") ?></a></h1></div>
<div class="entry-meta"><p><?php echo $page->getDateHtml("published") ?> <?php echo $this->yellow->language->getTextHtml("blogBy") ?> <?php $authorCounter = 0; foreach (preg_split("/\s*,\s*/", $page->get("author")) as $author) { if (++$authorCounter>1) echo ", "; echo htmlspecialchars($author); } ?></p></div>
<div class="entry-content"><?php echo $this->yellow->toolbox->createTextDescription($page->getContent(), 0, false, "<!--more-->", "<a href=\"".$page->getLocation(true)."\">".$this->yellow->language->getTextHtml("blogMore")."</a>") ?></div>
</div>
<?php endforeach ?>
</div>
</div>
<?php $this->yellow->layout("footer") ?>
```

Now add `Layout: home` to the settings at the top of your home page.

Tested with Datenstrom Yellow 0.8.19. Hope this helps.

**Comment by MrMacStripe** [comment]
Thank you very much, that´s exactly what I was looking for. Great support!

**Comment by markseuffert** [comment]
Glad I could help. I just realised that blog customisation was easier a few versions ago, probably have to look into it again. Let us know how Yellow works for you. Feedback is always welcome. :smile:

**Comment by MrMacStripe** [comment]
So far, it works inmsanly fast and is super easy. Exactly what I was looking for to get rid of wordpress since that is way to big for most of my projects. Your system provides basically anything that´s needed.

**Comment by kodersha** [comment]
It doesn't work anymore. Nothing is displayed.
